client/components/subcategories-questions.js

```diff
@@ -0,0 +1,51 @@
+import React, {Component} from 'react'
+
+class SubQuestions extends Component {
+  constructor(props) {
+    super(props)
+    this.handleClick = this.handleClick.bind(this)
+    this.state = {
+      isClosed: true
+    }
+  }
+
+  // handleClick = (event) => {
+  //   this.setState({
+  //     isClosed : !this.state.Closed
+  //   })
+  // }
+
+  handleClick(event) {
+    event.preventDefault()
+    this.setState(currentState => ({isClosed: !currentState.isClosed}))
+  }
+
+  render() {
+    let onChange = this.props.onChange
+    let subcategories = this.props.subcategories
+    return (
+      <div>
+        <button type="toggle" onClick={this.handleClick}>
+          Toggle to enter detailed breakdowns
+        </button>
+        {!this.state.isClosed
+          ? subcategories.map(current => {
+              return (
+                <div key={current.name}>
+                  <label htmlFor={current.name}>{current.name}</label>
+                  <input
+                    name={current.name}
+                    type="number"
+                    value={current.monthly}
+                    onChange={onChange}
```

Comment:
How is the subcategory data being processed by the parent component? Perhaps make a new method to pass down from the parent that only handles subcategory data to separate it from category data?

Reply:
Yeah, that was the issue that I was getting hung up on when I tried adding something to the state as a subcategory. I tried what you mentioned, but maybe I didn't set up the onChange function correctly b/c it didn't work... So, currently, when a user tries to enter numbers into the subcategory fields it adds it to the state as its own category.